src/autocomplete/src/AutocompleteItem.js

```diff
@@ -1,6 +1,98 @@
 import React, { memo, forwardRef } from 'react'
 import PropTypes from 'prop-types'
-import Option from '../../select-menu/src/Option'
+import { useListBehavior, useStyleConfig } from '../../hooks'
+import { Image } from '../../image'
+import { Pane } from '../../layers'
+import TextTableCell from '../../table/src/TextTableCell'
+
+const disableProps = { color: 'muted' }
+const selectedProps = { color: 'selected' }
+const emptyProps = {}
+
+const pseudoSelectors = {
+  _focus: ':focus',
+  _hover: ':hover',
+  _active: '&[aria-current="true"]:active, &[data-isselectable="true"]:active',
+  _current: '&[aria-current="true"]',
+  _isSelectable: '&[data-isselectable="true"]'
+}
+
+const internalStyles = {
+  display: 'flex',
+  alignItems: 'center'
+}
+
+/* eslint-disable react/prop-types */
+const Option = memo(
+  forwardRef(function Option(
+    {
+      disabled,
+      height,
+      icon,
+      isHighlighted,
+      isSelectable,
+      isSelected,
+      label,
+      onDeselect,
+      onSelect,
+      style,
+      ...rest
+    },
+    ref
+  ) {
+    let textProps = emptyProps
+    if (disabled) {
+      textProps = disableProps
+    }
+
+    if (isSelected) {
+      textProps = selectedProps
+    }
+
+    const { getRef, ...listBehaviorProps } = useListBehavior({
+      isSelectable,
+      isSelected,
+      disabled,
+      onSelect,
+      ref
+    })
```

Comment:
User-provided handlers will overwrite these `listBehaviorProps` based on how they are spread on the component below.